airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py

```diff
@@ -52,48 +58,51 @@ def send_request(self, query: str) -> SearchGoogleAdsResponse:
     @staticmethod
     def get_fields_from_schema(schema: Mapping[str, Any]) -> List[str]:
         properties = schema.get("properties")
-
         return [*properties]
 
     @staticmethod
-    def convert_schema_into_query(schema: Mapping[str, Any], report_name: str, from_date: str, to_date: str) -> str:
+    def convert_schema_into_query(
+        schema: Mapping[str, Any], report_name: str, from_date: str = None, to_date: str = None, cursor_field: str = None
+    ) -> str:
         from_category = REPORT_MAPPING[report_name]
         fields = GoogleAds.get_fields_from_schema(schema)
         fields = ",\n".join(fields)
 
-        query = Template(
-            """
-          SELECT
-            $fields
-          FROM $from_category
-          WHERE segments.date > '$from_date'
-            AND segments.date < '$to_date'
-          ORDER BY segments.date
-      """
-        )
-        query = query.substitute(fields=fields, from_category=from_category, from_date=from_date, to_date=to_date)
+        query_template = "SELECT {fields} FROM {from_category} "
+        substitute_params = dict(fields=fields, from_category=from_category)
+
+        if cursor_field:
+            query_template += "WHERE {cursor_field} > '{from_date}' AND {cursor_field} < '{to_date}' ORDER BY {cursor_field}"
+            substitute_params.update(dict(from_date=from_date, to_date=to_date, cursor_field=cursor_field))
 
-        return query
+        return query_template.format(**substitute_params)
 
     @staticmethod
-    def get_field_value(result: GoogleAdsRow, field: str) -> str:
+    def get_field_value(field_value: GoogleAdsRow, field: str) -> str:
         field_name = field.split(".")
-        try:
-            field_value = result
-            for level_attr in field_name:
-                field_value = field_value.__getattr__(level_attr)
-                if isinstance(field_value, Enum):
-                    field_value = field_value.name
+        for level_attr in field_name:
```

Comment:
could you add a comment explaining why this logic is necessary (and why the `_` logic is there as well) preferably with some URL or example of how it's needed? Not doubting that this is useful, just as a reviewer it is not clear why we are doing this, and it's not obvious to me where in the API docs to look to answer this quesiton

Reply:
added comments

Reply:
@sherifnada 
In the process of writing a comment, I found a way how we can simplify and improve data extraction, and get a nice structured object.
This is all done in one line, but there are a couple of nuances:
1. We will need to change the schemas for the new output (since the data will be nested according to their structure, now we get all the data to the very top).
2. It is necessary to hardcode the requested parameters (now we take them from the catalog)

I don’t think it is necessary to do this in this PR, but I can apply it while I’m working on Report Streams, I don’t think it will be long.
Thus, we get rid of these unnecessary checks and get a whole nested object.